src/plots/cartesian/layout_attributes.js

```diff
@@ -806,7 +806,20 @@ module.exports = {
         description: [
             'Sets the angle of the tick labels with respect to the horizontal.',
             'For example, a `tickangle` of -90 draws the tick labels',
-            'vertically.'
+            'vertically. If `tickangle` is auto, the angle will be chosen from',
+            'autotickangles if provided. If `autotickangles` is not provided,',
+            'the angle will be 30 or 90 degrees.'
+        ].join(' ')
+    },
+    autotickangles: {
+        valType: 'data_array',
```

Comment:
@alexcjohnson What's your thoughts here?

Reply:
Ah, we have an option that I'd forgotten about that's even better - see eg [splom xaxes/yaxes](https://github.com/plotly/plotly.js/blob/33dc2de5be0cdb7eb9ce51ad00283ec54b36f729/src/traces/splom/attributes.js#L36-L55):
```js
autotickangles: {
    valType: 'info_array',
    freeLength: true, // any length array is allowed
    items: { // with items not an array, all array entries are validated the same way
        valType: 'angle'
    }
}
```
This way you _must_ provide an array (which I think is the right call since the default is 2 items) but unlike `data_array` or `arrayOk` all items in the array are validated as `angle` items, ie numbers with nice bounds of [-180,180].

Reply:
I changed the valType to info_array as suggested!